crates/ra_lsp_server/src/main_loop.rs

```diff
@@ -587,17 +602,20 @@ fn update_file_notifications_on_threadpool(
     subscriptions: Vec<FileId>,
 ) {
     log::trace!("updating notifications for {:?}", subscriptions);
+    let publish_diagnostics = world.feature_flags().get("lsp.diagnostics");
     pool.execute(move || {
```

Comment:
Couldn't you do a pre-spawn condition check whether this will actually do something? This is in the main thread and loop, and maybe we don't even do something, yet we spawn a useless thread (iterating over files for nothing)?

Reply:
This doesn’t spawn a thread, just sends a closure over a channel, it”s only a constant amount of work, comparable to the work we need to get to this point in the first place.